soundcrate/src/app/user/[username]/profile/lists/page.js

```diff
@@ -28,21 +30,61 @@ export default function ListsPage({ params }) {
     fetchSongLists(username);
   }, [username]);
 
+  // auto updates
+  const handleNewList = (newList) => {
+    setLists([newList, ...lists]);
+  };
+
+  // deleting a list
+  const handleDeleteList = async (list_id) => {
+    try {
+      const response = await fetch(`/api/lists/deleteSongList?id=${list_id}`, {
+        method: 'DELETE',
+      });
+      if (response.ok) {
+        setLists(lists.filter(list => list._id !== list_id));
+      } else {
+        const responseData = await response.json();
+        console.error(responseData.error);
+      }
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
   const list_cards = (list_array) => {
+    if (!list_array || list_array.length === 0) {
+      return <EmptyContentMessage message="No Lists" />;
+    }
     return (list_array && list_array.map((list) =>
-      <ListCard 
-        key={list._id}
-        username={list.user.username}
-        list_id={list._id}
-        name={list.title}
-        song_count={list.songIds.length} 
-      />
+      <div key={list._id} className="flex items-center justify-between bg-dark-light p-4 rounded-md mb-4">
+        
+        <div className="flex-grow">
+          <ListCard 
+            username={list.user.username}
+            list_id={list._id}
+            name={list.title}
+            song_count={list.songIds.length} 
+          />
+        </div>
+        {session?.status === 'authenticated' && session?.user?.username === username && (
+          <button
+            className="btn p-2 bg-dark-light text-white rounded-md hover:bg-red"
+            onClick={() => handleDeleteList(list._id)}
+          >
+            Delete
+          </button>
```

Comment:
Nit, flagging that a good future update would be to make this a component and have it on an individual song list page

Reply:
i'll add this idea to our slides instead of this merge for right now